guard/src/rules/eval_context.rs

```diff
@@ -1174,22 +1175,74 @@ impl<'value, 'loc: 'value> EvalContext<'value, 'loc> for RootScope<'value, 'loc>
     }
 }
 
-pub(crate) fn validate_number_of_params(name: &str, num_args: usize) -> Result<()> {
-    let expected_num_args = match name {
-        "join" => 2,
-        "substring" | "regex_replace" => 3,
-        "count" | "json_parse" | "to_upper" | "to_lower" | "url_decode" | "parse_string"
-        | "parse_boolean" | "parse_float" | "parse_int" | "parse_char" => 1,
-        _ => {
-            return Err(Error::ParseError(format!(
-                "no such function named {name} exists"
-            )));
+#[derive(Debug)]
+pub(crate) enum Function {
```

Comment:
Nit: not 100% sure i agree with the naming here. I know its more verbose so maybe thats why you chose to just call it `Function` but I think `FunctionName` or something of that sort is more accurate

Reply:
Sounds good!